modules/workload-identity/main.tf

```diff
@@ -45,16 +44,45 @@ resource "google_service_account" "cluster_service_account" {
   project      = var.project_id
 }
 
-resource "kubernetes_service_account" "main" {
```

Comment:
https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/service_account_v1

Reply:
Thanks for the review @DrFaust92

I didn't realize there was another version of the resource, I will see how it behaves depending on the problem caused by the feature gate.

I will comment on the result and/or add the changes.

Reply:
I tried the resource you mentioned but it has the same behavior.